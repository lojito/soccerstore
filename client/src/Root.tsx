import { type ReactNode } from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { createStore } from "redux";
import reducers from "./reducers";

type Props = {
  children: ReactNode;
  initialStoreValue?: {
    favorites: number[];
    cart: number[];
  };
};

// eslint-disable-next-line react-refresh/only-export-components
export default ({ children, initialStoreValue }: Props) => {
  const store = createStore(
    reducers,
    initialStoreValue as unknown as Partial<{ favorites: never; cart: never }>
  );

  return (
    <BrowserRouter>
      <Provider store={store}>{children}</Provider>
    </BrowserRouter>
  );
};
