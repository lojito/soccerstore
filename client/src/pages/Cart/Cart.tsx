import { type FC } from "react";
import { useSelector } from "react-redux";
import { styled } from "styled-components";
import Product, { ProductType } from "../../components/Product/Product";
import { RootReducerType } from "../../reducers";
import { DivWrapper } from "../ProductList/ProductList";

export const EmptyCart = styled.div`
  text-align: center;
`;

type Props = {
  products: ProductType[];
  token: string;
};

const Cart: FC<Props> = ({ products, token }) => {
  const cartProducts = useSelector((state: RootReducerType) => state.cart);
  const filteredProducts = products.filter((product) => {
    return cartProducts.includes(product.id);
  });

  const renderProducts = filteredProducts.length ? (
    filteredProducts.map((product) => (
      <Product key={product.name} product={product} isLoggedIn token={token} />
    ))
  ) : (
    <EmptyCart>
      El carrito esta vacío. Continua tu compra de productos relacionados con el
      fútbol, el deporte más popular del mundo.
    </EmptyCart>
  );

  return <DivWrapper>{renderProducts}</DivWrapper>;
};

export default Cart;
