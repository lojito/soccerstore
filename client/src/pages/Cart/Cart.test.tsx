import { render } from "@testing-library/react";
import Root from "../../Root";
import Cart from "./Cart";

vi.mock("../../components/Product/Product");

describe("Cart", () => {
  it("should render correctly", () => {
    const products = [
      {
        id: 1,
        category: "balones",
        name: "Balon Blanco y Negro",
        price: "29.99",
        image: "1",
      },
      {
        id: 201,
        category: "espinilleras",
        name: "Espinilleras Negras",
        price: "12.17",
        image: "1",
      },
    ];
    const { asFragment } = render(
      <Root>
        <Cart products={products} token={"asfsdf"} />
      </Root>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("should display products in the cart", () => {
    const products = [
      {
        id: 1,
        category: "balones",
        name: "Balon Blanco y Negro",
        price: "29.99",
        image: "1",
      },
      {
        id: 201,
        category: "espinilleras",
        name: "Espinilleras Negras",
        price: "12.17",
        image: "1",
      },
    ];
    const { asFragment } = render(
      <Root initialStoreValue={{ favorites: [], cart: [1] }}>
        <Cart products={products} token={"asfsdf"} />
      </Root>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
