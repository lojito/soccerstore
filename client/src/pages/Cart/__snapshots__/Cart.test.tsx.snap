// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Cart > should display products in the cart 1`] = `
<DocumentFragment>
  <div
    class="sc-hLQSwg egqrsP"
  />
</DocumentFragment>
`;

exports[`Cart > should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-hLQSwg egqrsP"
  >
    <div
      class="sc-jTQCzO eAFyMX"
    >
      El carrito esta vacío. Continua tu compra de productos relacionados con el fútbol, el deporte más popular del mundo.
    </div>
  </div>
</DocumentFragment>
`;
